### ISSUE TITLE
Allow empty uri_segment without double slash

### ISSUE THREAD

**Issue by cbutcosk** [linked-art/crom#34]
In general crom's pattern of using the CRM-centric class names in entity URIs has been super useful--not having to worry about path components at model init is nice! But in some contexts I need to initialize models with an identifier I control, while still using `base_url` but without leaking CRM class names. AFAICT this isn't something that crom supports yet--perhaps I missed something? 

Sometimes I'd like URIs to have domain-appropriate terms (`/exhibition/1` rather than `/Acitivity/1`), and sometimes I'd like the URI to have a more complex relationship to its related components--(`/exhibition/1/venue/2`), etc, but I'd like those while still keeping the we'll-manage-id-path-components-for-you approach that makes model generation a breeze in crom.

Anyway should someone need this kind of thing in a pinch, we got around the issue using a wrapper for the initializer on `ExternalResource` that sets the id as we'd like after the class is done being initialized:
```python
def custom_ext_res_init(self,ident=None,_original=model.ExternalResource.__init__):
        """
        Intializes ExternalResource and then modifies self.id after init
        """
        _original(self,ident)

        if ident and ident.startswith('http'):
            self.id = ident
        elif ident:
            self.id = urllib.parse.urljoin( self._factory.base_url, ident )
        else:
            self.id = ''

    update_wrapper(custom_ext_res_init,model.ExternalResource.__init__)
    model.ExternalResource.__init__ = custom_ext_res_init  
```

**Comment by azaroth42** [comment]
Is there any alignment between class and the segment in the URI? The segment can be set per class (e.g. https://github.com/linked-art/crom/blob/master/cromulent/vocab.py#L839 ) but it's not more capable than that.

That said ... you can associate them with classes created dynamically from vocab...

```
>>> from cromulent import vocab
>>> vocab.Exhibition._uri_segment
'Activity'
>>> vocab.Exhibition._uri_segment = 'exhibition'
>>> x = vocab.Exhibition()
>>> x.id
'http://lod.example.org/museum/exhibition/0'
```
 
For the sub-paths, I do things like:

```
>>> a = model.Activity(ident=x.id + "/part/1")
>>> a.id
'http://lod.example.org/museum/exhibition/0/part/1'
```

But did you want an API something like `ident=(x, 'part')` and it would then apply the other rules to `part` for incrementing integer, uuid, etc ?

**Comment by cbutcosk** [comment]
On that specific project there was only sometimes an association of top-level class and URI pattern: eg, `/exhibition/1` and `/artist/1`, but `/person/1` (for non-artist `Person` instances) and `/organization/1` (for `Group` instances). And on other projects we've used the first path component to speak to source repository (eg, `/org1/humanmadeobject/1`) or other dataset partitions. So.. "sometimes?" 

Anyway you're of course right that _uri_segment is a happier integration path. IIRC the trouble was that an empty `_uri_segment` will emit a double slash, allowing which would would fix my issue, eg:
```python
>>> from cromulent import model, vocab
>>> vocab.Exhibition._uri_segment = ""
>>> exh = vocab.Exhibition(ident="org1/exhibition/2")
>>> exh.id
'http://lod.example.org/museum//org1/exhibition/2'
```

**Comment by azaroth42** [comment]
Ahha! Yes, that's an easy bug to fix :) Will do that ASAP

**Comment by azaroth42** [comment]
```
>>> model.Person._uri_segment = ""
>>> p = model.Person()
>>> p.id
'http://lod.example.org/museum/0'
>>> model.Person._uri_segment = "foo"
>>> p = model.Person()
>>> p.id
'http://lod.example.org/museum/foo/0'
```

Available in 0.16.9 on pypi

**Comment by cbutcosk** [comment]
Sorry to bother you with this but it looks like the doubled slash still shows up with an `ident` argument:
```python
>>> p = model.Person(ident="vocab1/person/1")
>>> p.id
'http://lod.example.org/museum//vocab1/person/1'
```

**Comment by azaroth42** [comment]
Fixed in 16.10
